drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -129,14 +129,72 @@ zephyr_compile_definitions_ifdef(CONFIG_NRF700X_ON_QSPI
 )
 
 # RPU FW patch binaries based on the selected configuration
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SYSTEM_MODE
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin
+if(CONFIG_NRF700X_SYSTEM_MODE)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin)
+elseif (CONFIG_NRF700X_RADIO_TEST)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin)
+elseif (CONFIG_NRF700X_SCAN_ONLY)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin)
+else()
+	# Error
+	message(FATAL_ERROR "Unsupported nRF70 patch configuration")
+endif()
+
+zephyr_compile_definitions(
+	-DCONFIG_NRF_WIFI_FW_BIN=${NRF70_PATCH}
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_RADIO_TEST
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin
+if (CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE)
+message(STATUS "nRF WiFi FW patch binary will be stored in external flash")
+
+if (CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
+set(NRF_QSPI_FLASH_BASE 0x10000000)
+elseif(CONFIG_BOARD_NRF52840DK_NRF52840)
+set(NRF_QSPI_FLASH_BASE 0x12000000)
+else()
+message(FATAL_ERROR "Unsupported board: ${BOARD}")
+endif()
+
+# Should be same as nrf70_fw_partition
+set(NRF70_PATCH_OFFSET 0x0)
+math(EXPR NRF70_PATCH_ADDR "${NRF_QSPI_FLASH_BASE} + ${NRF70_PATCH_OFFSET}")
+
+find_package(Python3)
+# detect virtualenv and set Pip args accordingly
+if(DEFINED ENV{VIRTUAL_ENV})
+  set(_pip_args)
+else()
+  set(_pip_args "--user")
+endif()
+
+# Check for pip show intelhex
+execute_process(
+	COMMAND ${Python3_EXECUTABLE} -m pip show intelhex
+	RESULT_VARIABLE _pip_show_result
+	OUTPUT_QUIET
+	ERROR_QUIET
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SCAN_ONLY
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin
+if(NOT _pip_show_result EQUAL 0)
+	message(FATAL_ERROR "intelhex Python package not found. Install it with: pip install ${_pip_args} intelhex")
+endif()
+
+# bin2hex.py is used to convert the binary patch to hex format
+execute_process(
+	COMMAND bin2hex.py --offset=${NRF70_PATCH_ADDR} ${NRF70_PATCH} ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex
```

Comment:
AFAIK, this will not work on windows as windows does not process shebang lines. Also, we should always use `Python3_EXECUTABLE` to make sure the right Python version is chosen. That may require that you provide a custom script, such as bin2hex.py unless there's already something in Zephyr for converting bin 2 hex...

Reply:
Ah ok, I will fix this.

Reply:
This is a bit tricky as `bin2hex.py` can be installed anywhere, here it is expected to be in $PATH, so, cannot invoke using `python3` without adding the dependency of the absolute path. Let me test this on Windows and see if it works as long as `bin2hex.py` is in PATH, windows should invoke with python based on extension. And as for the right python, I don't think we can do much, only user can control it.

Reply:
If it doesn't work on windows, you can always do sth like:
```
COMMAND ${Python3_EXECUTABLE}
    -c "import sys; import intelhex; intelhex.bin2hex(sys.argv[1], sys.argv[2], int(sys.argv[3]))"
    ${NRF70_PATCH}
    ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex
    ${NRF70_PATCH_ADDR}
```

Reply:
It didn't work on windows even though `bin2hex.py` was in the path, so, went with your solution, thanks.